driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotCreateMongocryptdClientTest.java

```diff
@@ -0,0 +1,224 @@
+    private MongoCollection<Document> collection;
+    private ConnectionTracker mongocryptdConnectionTracker;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeTrue(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null);
+        mongocryptdConnectionTracker = ConnectionTracker.start();
+        client = createMongoClient(MongoClientSettings.builder(getMongoClientSettings())
+                .autoEncryptionSettings(AutoEncryptionSettings.builder()
+                        .kmsProviders(singletonMap(LOCAL_KMS_PROVIDER_ID, singletonMap("key", localKmsProviderKey())))
+                        .keyVaultNamespace(KEY_VAULT_NAMESPACE.getFullName())
+                        .extraOptions(Stream.of(
+                                new SimpleImmutableEntry<>("cryptSharedLibPath", CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE),
+                                new SimpleImmutableEntry<>("mongocryptdURI", format("mongodb://%s:%d/db?serverSelectionTimeoutMS=%d",
+                                        mongocryptdConnectionTracker.serverSocket().getInetAddress().getHostAddress(),
+                                        mongocryptdConnectionTracker.serverSocket().getLocalPort(),
+                                        TIMEOUT.toMillis()))
+                        ).collect(toMap(Entry::getKey, Entry::getValue)))
+                        .build())
+                .build());
+        client.getDatabase(KEY_VAULT_NAMESPACE.getDatabaseName()).drop();
+        MongoDatabase db = client.getDatabase("db");
+        db.drop();
+        collection = db.getCollection("coll");
+    }
+
+    @AfterEach
+    @SuppressWarnings("try")
+    public void cleanUp() throws Exception {
+        //noinspection unused
+        try (ConnectionTracker autoClosed = mongocryptdConnectionTracker;
+             MongoClient autoClosed2 = client) {
+            // empty
+        }
+    }
+
+    protected abstract MongoClient createMongoClient(MongoClientSettings settings);
+
+    @Test
+    @SuppressWarnings("try")
+    void whenCryptSharedLoaded() throws Exception {
+        //noinspection unused
+        try (AutoCloseable assertNoConnectionsOnAutoCloseToPreserveAssertionFailure = mongocryptdConnectionTracker) {
+            collection.insertOne(Document.parse("{unencrypted: 'test'}"));
+        }
+    }
+
+    static int findAvailableMongocryptdLoopbackPort() {
+        try (ServerSocket serverSocket = new ServerSocket()) {
+            serverSocket.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            int foundPort = serverSocket.getLocalPort();
+            if (foundPort != DEFAULT_MONGOCRYPTD_PORT) {
+                return foundPort;
+            } else {
+                return findAvailableMongocryptdLoopbackPort();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to find an available port", e);
+        }
+    }
```

Comment:
We could have just used `serverSocket.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0))` right away in `ConnectionTracker.start` instead of introducing the `findAvailableMongocryptdLoopbackPort` method. However, the same approach is not helpful for the tests in `AbstractClientSideEncryptionNotSpawnMongocryptdTest`. Additionally, making sure that we never use the default `mongocryptd` port in `AbstractClientSideEncryptionNotSpawnMongocryptdTest` also improve the isolation between different tests.

To avoid using different approaches, I simply use `findAvailableMongocryptdLoopbackPort` in both `AbstractClientSideEncryptionNotCreateMongocryptdClientTest` and `AbstractClientSideEncryptionNotSpawnMongocryptdTest.